### PR TITLE
Avoid thread.detach(), join when needed instead

### PR DESCRIPTION
**Pull request by unknownbrackets**
This should help #12323, because the Switch doesn't currently have a thread implementation that supports `detach()`.  So this largely avoids it (though maybe I missed some.)  FYI @m4xw.

There are a few changes in behavior here:

1. Some things will join on game exit now, which could cause a delay in exit that wasn't there before.
2. Reporting now serializes messages, rather than a new thread for each report.  This will probably reduce perf impact, but also report more slowly.  It tries to abort on game exit.

Tried to test each of the parts I changed, but there's probably some risk I missed a join and it might std::system_error.

-[Unknown]

### CODE REVIEW

**Comment by hrydgard** [comment]
I'm fine with all of this. I do feel that use of joinable() is a bit of a code smell - if you need to check, your design can probably be improved, but no big deal. 

Time to merge and let any bugs shake out.

**Comment by unknownbrackets** [comment]
In some cases it was an alternative to adding a flag of whether it was started or not.  I'm assuming it's just a flag in STL.  That said, some could likely be removed without adding a flag, was also playing it safe.

-[Unknown]
